### PR TITLE
php, php-cgi_select: add port select for php-cgi

### PR DESCRIPTION
**Pull request by jandemter**
#### Description

adds 'port select' for 'php-cgi'

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.2.1 20D74
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/php/Portfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ PortSystem              1.0
 PortGroup               php 1.1
 PortGroup               compiler_blacklist_versions 1.0
 PortGroup               old_openssl 1.0
+PortGroup               select 1.0
```
Comment (ryandesign) [inline lang/php/Portfile:7]: If we go with your proposal, then still not all of the subports need the select portgroup. Assuming that including the portgroup but not using it is not harmful, we would still probably want to restrict it to only a reasonable subset of the subports, such as the existing block handling all SAPI subports.

**Review by ryandesign — COMMENTED** [review]
Do we really want to be able to select php-cgi separately from php, or should it be part of selecting php? The latter was proposed in https://trac.macports.org/ticket/45999. In some php versions there is also a php-cgi manpage which the patch in that ticket also handles.

I also wondered aloud in that ticket whether there is any benefit to including apache2handler and fpm in the select mechanism as well.

File: lang/php/Portfile (hunk 1 above)
Comment (jandemter) [reply lang/php/Portfile]: I force pushed a change to only use the select group for php and php-cgi. The motivation for the change is that a certain PHP IDE will try to use php-cgi for the integrated dev server no matter what.

**Comment by reneeotten** [comment]
@ryandesign do you want to move forward with this PR?

**Comment by reneeotten** [comment]
please re-open once there is a consensus on what to do there - closing due to inactivity

**Pull request closed without merging**